nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/AbstractBaseAWSProcessor.java

```diff
@@ -0,0 +1,211 @@
+                .required(false)
+                .identifiesControllerService(SSLContextService.class)
+                .build();
+    public static final PropertyDescriptor ENDPOINT_OVERRIDE = new PropertyDescriptor.Builder()
+                .name("Endpoint Override URL")
+                .description("Endpoint URL to use instead of the AWS default including scheme, host, port, and path. " +
+                        "The AWS libraries select an endpoint URL based on the AWS region, but this property overrides " +
+                        "the selected endpoint URL, allowing use with other S3-compatible endpoints.")
+                .required(false)
+                .addValidator(StandardValidators.URL_VALIDATOR)
+                .build();
+    public static final PropertyDescriptor PROXY_HOST = new PropertyDescriptor.Builder()
+            .name("Proxy Host")
+            .description("Proxy host name or IP")
+            .expressionLanguageSupported(true)
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor PROXY_HOST_PORT = new PropertyDescriptor.Builder()
+            .name("Proxy Host Port")
+            .description("Proxy host port")
+            .expressionLanguageSupported(true)
+            .required(false)
+            .addValidator(StandardValidators.PORT_VALIDATOR)
+            .build();
+
+    protected volatile Region region;
+    protected static final Protocol DEFAULT_PROTOCOL = Protocol.HTTPS;
+    protected static final String DEFAULT_USER_AGENT = "NiFi";
+
+    private static AllowableValue createAllowableValue(final Regions regions) {
+        return new AllowableValue(regions.getName(), regions.getName(), regions.getName());
+    }
+
+    private static AllowableValue[] getAvailableRegions() {
+        final List<AllowableValue> values = new ArrayList<>();
+        for (final Regions regions : Regions.values()) {
+            values.add(createAllowableValue(regions));
+        }
+
+        return (AllowableValue[]) values.toArray(new AllowableValue[values.size()]);
+    }
+
+    public AbstractBaseAWSProcessor() {
+        super();
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSessionFactory sessionFactory) throws ProcessException {
+        final ProcessSession session = sessionFactory.createSession();
+        try {
+            onTrigger(context, session);
+            session.commit();
+        } catch (final Throwable t) {
+            getLogger().error("{} failed to process due to {}; rolling back session", new Object[]{this, t});
+            session.rollback(true);
+            throw t;
+        }
+    }
```

Comment:
I need access to the ProcessSessionFactory to handle asynchronous event processing by KCL.  Please see my notes below and let me know if there is a better way to handle this.